### PR TITLE
CEDS-1257 Review Arrive & Depart Validation

### PR DESCRIPTION
**Pull request by PauloPortugal**
* Validate EORI and MUCR/DUCR
* Validate two letter country codes

### CODE REVIEW

File: test/forms/ConsignmentReferencesSpec.scala [hunk 1]
```diff
@@ -42,28 +42,46 @@ class ConsignmentReferencesSpec extends BaseSpec {
...
     }
 
     "return error for incorrect reference" in {
-      val inputData = ConsignmentReferences(Some("GB71757250450281160"), "Incorrect reference", "12345")
+      val inputData =
+        ConsignmentReferences(Some("GB71757250450281160"), "Incorrect reference", "5123456789-000-123ABC45$%^FIIIII")
       val errors = ConsignmentReferences.form().fillAndValidate(inputData).errors
 
-      errors.length must be(2)
+      errors.length must be(3)
```
Comment (benjaminparker) [inline test/forms/ConsignmentReferencesSpec.scala]: Assert this in a single line.  errors mustBe List(....)

File: test/forms/ConsignmentReferencesSpec.scala [hunk 2]
```diff
@@ -42,28 +42,46 @@ class ConsignmentReferencesSpec extends BaseSpec {
...
       errors(1) must be(FormError("reference", "consignmentReferences.reference.error"))
+      errors(2) must be(FormError("referenceValue", "consignmentReferences.reference.value.error"))
     }
 
-    "no errors when data is correct" in {
-      val inputData = ConsignmentReferences(Some("GB717572504502811"), "D", "123456")
-      val errors = ConsignmentReferences.form().fillAndValidate(inputData).errors
+    "no errors when data is correct with different valid MUCR/DUCRs" in {
+      val inputData1 = ConsignmentReferences(Some("GB717572504502811"), "D", "5GB123456789000-123ABC456DEFIIIII")
+      val errors1 = ConsignmentReferences.form().fillAndValidate(inputData1).errors
```
Comment (benjaminparker) [inline test/forms/ConsignmentReferencesSpec.scala]: use mustBe empty

File: test/forms/ConsignmentReferencesSpec.scala [hunk 3]
```diff
@@ -42,28 +42,46 @@ class ConsignmentReferencesSpec extends BaseSpec {
...
+      errors1.length must be(0)
+
+      val inputData2 = ConsignmentReferences(Some("GB717572504502811"), "D", "GB/ABC4-ASIUDYFAHSDJF")
+      val errors2 = ConsignmentReferences.form().fillAndValidate(inputData2).errors
+      errors2.length must be(0)
```
Comment (benjaminparker) [inline test/forms/ConsignmentReferencesSpec.scala]: use mustBe empty

File: app/utils/validators/forms/FieldValidator.scala [hunk 4]
```diff
@@ -44,6 +46,12 @@ object FieldValidator {
     def or(second: Boolean): Boolean = first || second
   }
 
+  val validEori: String => Boolean = (input: String) => input.matches("^GB[0-9]{1,15}$")
```
Comment (benjaminparker) [inline app/utils/validators/forms/FieldValidator.scala]: We could use tiny types like case class Eori(val: String) extends AnyVal.  Then put the valid Regex with the class.  Then we don't have the possibility of another String ending up there and we have the validation with the thing being validated.  An easy fix that would save a lot of bugs ?

Comment (PauloPortugal) [reply app/utils/validators/forms/FieldValidator.scala]: I am a fan of dynamic typed functional languages :)
This function solves the problem at hand.

File: test/forms/ConsignmentReferencesSpec.scala (hunk 3 above)
Comment (PauloPortugal) [reply test/forms/ConsignmentReferencesSpec.scala]: Addressed!

File: test/forms/ConsignmentReferencesSpec.scala (hunk 2 above)
Comment (PauloPortugal) [reply test/forms/ConsignmentReferencesSpec.scala]: Addressed!

File: test/forms/ConsignmentReferencesSpec.scala (hunk 1 above)
Comment (PauloPortugal) [reply test/forms/ConsignmentReferencesSpec.scala]: Addressed!

**Pull request merged**